File: Python_Codes/14. Object Class Methods/ToString.py
# Class representing a Person
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    # Overriding the __str__ method
    def __str__(self):
        return f"Person(name='{self.name}', age={self.age})"

# Main logic to demonstrate __str__()
def main():
    person = Person("John Doe", 30)

    # Printing the object directly
    # The overridden __str__ method is called automatically
    print(person)

if __name__ == "__main__":
    main()

File: Java_Codes/14. Object Class Methods/ToString.java
// Class representing a Person
class Person {
    private String name;
    private int age;

    // Constructor to initialize the Person object
    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    // Overriding the toString() method
    @Override
    public String toString() {
        return "Person{name='" + name + "', age=" + age + "}";
    }
}

// Main class to demonstrate the toString() method
public class ToString {
    public static void main(String[] args) {
        // Creating a Person object
        Person person = new Person("John Doe", 30);

        // Printing the object directly
        // The overridden toString() method is called automatically
        System.out.println(person);
    }
}